### PR TITLE
Add Chrome Custom Tabs/Safari View Controller (replace internal browser)

### PR DESCRIPTION
**Pull request by swmarks**
I didn't love the in-app view we had been using, so I added Chrome Custom Tabs for Android and Safari View Controller for iOS. This should maintain compatibility across OS's through conditional statements, but testing is welcome.

I also changed the nomenclature for the settings page to be "Use external browser:" and flipped the logic because they're both external in a way.

### CODE REVIEW

**Comment by zachatrocity** [comment]
Could you add a screenshot?

**Comment by swmarks** [comment]
> Could you add a screenshot?

Yeah, unfortunately GitHub is not letting me add videos at the moment, but here's the settings:

Screenshot_20230703_083140

And here's what it looks like. Just a Chrome Custom Tab web view with the page title and options to open in an external browser from there if you want. It follows our primaryColor:

Screenshot_20230703_083203_Vivaldi Browser

**Comment by swmarks** [comment]
I merged @mykdavies translations which reverted my flipping the logic from "Use in app browser" to "Use external browser" in settings. I actually think I'm going to leave it this way for now because I think if we want to change the naming convention for that settings toggle, we should do it the right way throughout the project, not the hacky way by just flipping the logic at the front end. I will make it happen if we want to go through with that change, but don't want to waste time if we don't.

File: lib/url_launcher.dart [hunk 1]
```diff
@@ -11,6 +14,40 @@ import 'pages/user.dart';
...
+        preferredControlTintColor:
+            textColorBasedOnBackground(Theme.of(context).colorScheme.primary),
+        barCollapsingEnabled: true,
+        entersReaderIfAvailable: false,
+        dismissButtonStyle: SafariViewControllerDismissButtonStyle.close,
+      ),
+    );
+  } catch (e) {
+    // An exception is thrown if browser app is not installed on Android device.
+    debugPrint(e.toString());
```
Comment (shocklateboy92) [inline lib/url_launcher.dart:48]: I see this in the example code of the library. 
Could you please convert this to show the snack bar like we do for other errors.

Or maybe fallback to external browser?

**Review by shocklateboy92 — CHANGES_REQUESTED** [review]
OMG YES! 
This is something I've been trying to do for a while, but didn't realize there was a library that did all the work for me.

Could you please update the error handling? Very excited to have this feature.

File: lib/url_launcher.dart (hunk 1 above)
Comment (swmarks) [reply lib/url_launcher.dart]: Definitely, implemented both suggestions in theory even if I don't have a great way of testing it without uninstalling/disabling all my browsers.

**Review by shocklateboy92 — APPROVED** [review]

**Pull request merged**